File: prime.py
import math

def is_prime(num):
    """
    Checks if a given number is prime.
    A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself.
    """
    if num <= 1:
        return False  # Numbers less than or equal to 1 are not prime

    # Iterate from 2 up to the square root of the number
    # If a number has a divisor greater than its square root, it must also have a divisor smaller than its square root.
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False  # If divisible by any number in this range, it's not prime
    return True  # If no divisors found, it's prime

# Iterate from 1 to 100 and print prime numbers
print("Prime numbers from 1 to 100:")
for number in range(1, 101):
    if is_prime(number):
        print(number)

File: prime.java
public class Prime 
{
public static void main(String[] args) 
{
System.out.println("Prime numbers from 1 to 100 are:");
for (int num = 2; num <= 100;) 
{
if (isPrime(num)) 
{
System.out.print(num + " ");
}
}
}
// Function to check if a number is prime
public static boolean isPrime(int num) 
{
if(num <= 1)
return false;
for(int i= 2; i <= math.sqrt(n); i++)
{
if (n % i == 0)
return false; 
}
return true;
}
}